hudi-gcp/src/main/java/org/apache/hudi/gcp/bigquery/HoodieBigQuerySyncClient.java

```diff
@@ -328,6 +323,29 @@ public boolean tableNotExistsOrDoesNotMatchSpecification(String tableName) {
     return manifestDoesNotExist;
   }
 
+  private boolean isBasePathUpdated(ExternalTableDefinition externalTableDefinition) {
+    String basePath = StringUtils.stripEnd(getBasePath(), "/");
+    if (externalTableDefinition.getHivePartitioningOptions() == null) {
+      List<String> sourceUris = Option.ofNullable(externalTableDefinition.getSourceUris()).orElse(Collections.emptyList());
```

Comment:
What's the scenario when `externalTableDefinition.getSourceUris()` contains multiple URIs?

Reply:
Not exactly sure about the scenario, but given the flexibility on this data model, i've added the logic to handle multiple URIs in `externalTableDefinition.getSourceUris()`.